hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenAirdropTest.java

```diff
@@ -1009,53 +1012,157 @@ final Stream<DynamicTest> nftWithRoyaltyFeesPaidByReceiverFails() {
+                            .hasTokenBalance(DENOM_TOKEN, 3 * HTS_FEE),
+                    withOpContext((spec, log) ->
+                            Assertions.assertEquals(currentCollectorBalance.get(), newCollectorBalance.get())),
+                    validateChargedUsd("FT with HTS fee airdrop to collector", 0.002, 20));
         }
 
         @HapiTest
         @DisplayName("NFT with royalty fee with treasury as receiver")
         final Stream<DynamicTest> nftWithRoyaltyFeesPaidByReceiverWithTreasuryReceiver() {
+            // declare treasury account balance variables
+            final AtomicLong currentTreasuryBalance = new AtomicLong();
+            final AtomicLong newTreasuryBalance = new AtomicLong();
             return hapiTest(
-                    cryptoCreate(OWNER),
+                    // set initial treasury balance variable
+                    getAccountBalance(TREASURY_FOR_CUSTOM_FEE_TOKENS).exposingBalanceTo(currentTreasuryBalance::set),
+                    cryptoCreate(OWNER).balance(initialBalance),
                     tokenAssociate(OWNER, NFT_WITH_ROYALTY_FEE),
                     cryptoTransfer(
                             movingUnique(NFT_WITH_ROYALTY_FEE, 3L).between(TREASURY_FOR_CUSTOM_FEE_TOKENS, OWNER)),
                     tokenAirdrop(movingUnique(NFT_WITH_ROYALTY_FEE, 3L).between(OWNER, TREASURY_FOR_CUSTOM_FEE_TOKENS))
-                            .signedByPayerAnd(TREASURY_FOR_CUSTOM_FEE_TOKENS, OWNER));
-        }
-
-        @HapiTest
-        @DisplayName("FT with royalty fee with treasury as receiver")
+                            .signedByPayerAnd(TREASURY_FOR_CUSTOM_FEE_TOKENS, OWNER)
+                            .payingWith(OWNER)
+                            .via("NFT with royalty fee airdrop to treasury"),
+                    // set new treasury balance variable
+                    getAccountBalance(TREASURY_FOR_CUSTOM_FEE_TOKENS)
+                            .exposingBalanceTo(newTreasuryBalance::set)
+                            .hasTokenBalance(NFT_WITH_ROYALTY_FEE, 99),
+                    // assert owner balance
+                    withOpContext((spec, log) -> {
+                        final var record = getTxnRecord("NFT with royalty fee airdrop to treasury");
+                        allRunFor(spec, record);
+                        final var txFee = record.getResponseRecord().getTransactionFee();
+                        // the transaction fee should be charged
+                        final var ownerBalance = getAccountBalance(OWNER)
+                                .hasTinyBars(initialBalance - txFee)
+                                .hasTokenBalance(NFT_WITH_ROYALTY_FEE, 0);
+                        allRunFor(spec, ownerBalance);
+                    }),
+                    // assert treasury balance is not changed
+                    withOpContext((spec, log) ->
+                            Assertions.assertEquals(currentTreasuryBalance.get(), newTreasuryBalance.get())),
```

Comment:
You can put this assertion in the withOpContext block above instead of creating new withOpContext

Reply:
Thanks for the hint, fixed in the latest commit.